### ISSUE TITLE
Improve exception when multiple persistences are configured

### ISSUE THREAD

**Issue by dbelcham** [Particular/NServiceBus#5261]
When a user configures multiple persistences, even if they are for different purposes, an exception is thrown. That exception is not clear about the root cause of the problem.

```
endpointConfiguration.UsePersistence<NHibernatePersistence>();
endpointConfiguration.UsePersistence<InMemoryPersistence,StorageType.Sagas>();
```

When run it blows up with
```
System.InvalidCastException occurred
  HResult=0x80004002
  Message=Unable to cast object of type 'NServiceBus.Persistence.NHibernate.NHibernateLazyNativeTransactionSynchronizedStorageSession' to type 'NServiceBus.InMemorySynchronizedStorageSession'.
  Source=NServiceBus.Core
  StackTrace:
   at NServiceBus.InMemorySagaPersister.Save(IContainSagaData sagaData, SagaCorrelationProperty correlationProperty, SynchronizedStorageSession session, ContextBag context) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Persistence\InMemory\SagaPersister\InMemorySagaPersister.cs:line 77
```
This situation should present the user with a meaningful (custom) exception that guides them to a solution.

**Comment by SzymonPobiega** [comment]
In `SQLP the `StorageSessionFeature` class is responsible for validating if both saga and outbox use SQLP. In order do do so it queries the enabled features API for two specific SQLP features.

Because there is no explicit seam for the saga/outbox persister, I am afraid it would be hard to handle this problem generically.

**Comment by SzymonPobiega** [comment]
Fixed in #5435
